src/style.rs

```diff
@@ -674,12 +678,96 @@ enum Alignment {
     Right,
 }
 
+/// Trait for defining stateful or stateless formatters
+pub trait ProgressTracker: Send {
+    fn cloned(&self) -> Box<dyn ProgressTracker>;
+    fn tick(&mut self, state: &ProgressState, now: Instant);
+    fn reset(&mut self, state: &ProgressState, now: Instant);
+    fn write(&self, state: &ProgressState, w: &mut dyn fmt::Write);
+}
+
+impl Clone for Box<dyn ProgressTracker>
+where
+    Self:,
+{
+    fn clone(&self) -> Self {
+        self.cloned()
+    }
+}
+
+impl<F: Fn(&ProgressState, &mut dyn fmt::Write) + Send + Clone + 'static> ProgressTracker for F {
```

Comment:
Like this? I figured not having a return at all is a better option as it lets the user decide how they want to handle any failures rather than allowing a potential panic in the crate.

Reply:
Yes, this seems reasonable.